ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -853,6 +887,76 @@ void processServerError(ServerErrorCode errorCode) {
           break;
       }
     }
+
+    /**
+     * Process a metadata blob to find the data chunks that need to be fetched.
+     * @param blobData the metadata blob's data.
+     * @throws IOException
+     * @throws MessageFormatException
+     */
+    private void handleMetadataBlob(BlobData blobData) throws IOException, MessageFormatException {
+      ByteBuffer serializedMetadataContent = blobData.getStream().getByteBuffer();
+      CompositeBlobInfo compositeBlobInfo =
+          MetadataContentSerDe.deserializeMetadataContentRecord(serializedMetadataContent, blobIdFactory);
+      chunkSize = compositeBlobInfo.getChunkSize();
+      totalSize = compositeBlobInfo.getTotalSize();
+      List<StoreKey> keys = compositeBlobInfo.getKeys();
+      if (options != null && options.getRange() != null) {
+        try {
+          resolvedByteRange = options.getRange().toResolvedByteRange(totalSize);
+        } catch (IllegalArgumentException e) {
+          onInvalidRange("ByteRange " + options.getRange() + " is not within total blob size " + totalSize);
+          return;
```

Comment:
is there any way you could get rid of this `return`? Nested `return`s that aren't at the beginning of a function can make it hard to understand the flow.